src/binfhe/lib/binfhecontext.cpp

```diff
@@ -42,92 +42,214 @@ void BinFHEContext::GenerateBinFHEContext(uint32_t n, uint32_t N,
                                           const NativeInteger &Q, double std,
                                           uint32_t baseKS, uint32_t baseG,
                                           uint32_t baseR, BINFHEMETHOD method) {
-  auto lweparams = std::make_shared<LWECryptoParams>(n, N, q, Q, std, baseKS);
+  auto lweparams = std::make_shared<LWECryptoParams>(n, N, q, Q, Q, std, baseKS);
   m_params =
       std::make_shared<RingGSWCryptoParams>(lweparams, baseG, baseR, method);
 }
 
+
+
+void BinFHEContext::GenerateBinFHEContext(BINFHEPARAMSET set,
+                                          bool arbFunc, uint32_t logQ, long N,  
+                                          BINFHEMETHOD method, bool timeOptimization) {
+  
+  if(method == AP){
```

Comment:
Is there a particular reason this fails here? Seems like `AP` doesn't get checked in the other instances of this function?

Reply:
Yes that's on purpose. Only for the case of large-precision sign evaluation / digit decomposition / arbitrary LUT evaluation, AP is not supported. And for these functionalities, this function for parameter generation should be called. But probably should add an explanation in README for that?

Reply:
Oohh, yeah, that's definitely good to have. I don't have the knowledge to write it up well do you have the bandwidth to do so?

Reply:
Absolutely. I will write that down in the README later today. Thanks!